source/CommonJobs/CommonJobs.MVC.UI/Areas/Evaluations/Controllers/EvaluationsApiController.cs

```diff
@@ -137,10 +139,44 @@ public JsonNetResult DeletePeriod(string period)
 
         [AcceptVerbs(HttpVerbs.Post)]
         [CommonJobsAuthorize(Roles = "EvaluationManagers")]
-        public JsonNetResult ChangeResponsible (string evaluatedUserName, string period, string newResponsibleName)
+        public JsonNetResult ChangeResponsible(string evaluatedUserName, string period, string newResponsibleName)
         {
             ExecuteCommand(new ChangeResponsibleCommand(evaluatedUserName, period, newResponsibleName));
             return Json("OK");
         }
+
+        [AcceptVerbs(HttpVerbs.Post)]
+        [CommonJobsAuthorize(Roles = "EmployeeManagers")]
+        public JsonNetResult RevertEvaluationState(string period, string username, string operation)
+        {
+            ExecuteCommand(new RevertEvaluationStateCommand(period, username, operation));
+            return Json("OK");
+        }
+
+        [AcceptVerbs(HttpVerbs.Get)]
+        [CommonJobsAuthorize(Roles = "EmployeeManagers")]
+        public JsonNetResult GetPosibleActions(string period, string userName)
+        {
+            var evaluationId = EmployeeEvaluation.GenerateEvaluationId(period, userName);
+            var evaluation = RavenSession.Load<EmployeeEvaluation>(evaluationId);
+            var califications = RavenSession.Advanced.LoadStartingWith<EvaluationCalification>(evaluationId + "/").ToList();
+            var isResponsibleEvalFinished = califications.Any(x => x.Owner == CalificationType.Responsible && x.Finished);
+            var isCompanyEvalFinished = califications.Any(x => x.Owner == CalificationType.Company && x.Finished);
+            var isAutoEvalFinished = califications.Any(x => x.Owner == CalificationType.Auto && x.Finished);
+            var isEvaluatorEvalFinished = califications.Any(x => x.Owner == CalificationType.Evaluator && x.Finished);
+            var actions = GetPosibleRevertActions(
```

Comment:
`GetPosibleRevertActions` is being used when you get the evaluations list, and also when you _revert an evaluation_, why is it need a third time?